packages/manager/test/SliceMachineManager-customTypes-updateCustomType.test.ts

```diff
@@ -40,3 +44,295 @@ it("throws if plugins have not been initialized", async (ctx) => {
+			const { crId, ids } = args ?? {};
+
+			return ctx.mockPrismic.model.customType({
+				format: "custom",
+				label: "Test CT",
+				repeatable: false,
+				status: true,
+				id: "testCt",
+				fields: {
+					name: {
+						type: "Text",
+						config: { label: "Name", placeholder: "Type name" },
+					},
+					authorDetails: {
+						type: "Link",
+						config: {
+							label: "Author Details",
+							placeholder: "Select author",
+							select: "document",
+							customtypes: [
+								{
+									id: "author",
+									fields: [
+										"authorFirstName",
+										{
+											id: crId ?? "address_cr",
+											customtypes: [
+												{
+													id: "address",
+													fields: ["country", ...(ids ?? [])],
+												},
+											],
+										},
+									],
+								},
+							],
+						},
+					},
+				},
+			});
+		};
+
+		const onUpdate = vi.fn();
+		updateCustomTypeContentRelationships({
+			models: [
+				{ model: getTwoLevelCrModel({ ids: ["city"] }) },
+				{ model: getTwoLevelCrModel({ ids: ["addressLine1"] }) },
+				{ model: getTwoLevelCrModel({ ids: ["addressLine1", "city"] }) },
+			],
+			previousPath: ["address", "city"],
+			newPath: ["address", "city_CHANGED"],
+			onUpdate,
+		});
+
+		expect(onUpdate).toHaveBeenCalledWith(
+			getTwoLevelCrModel({ ids: ["city_CHANGED"] }),
+		); // changed
+		expect(onUpdate).toHaveBeenCalledWith(
+			getTwoLevelCrModel({ ids: ["addressLine1"] }),
+		); // not changed
+		expect(onUpdate).toHaveBeenCalledWith(
+			getTwoLevelCrModel({ ids: ["addressLine1", "city_CHANGED"] }),
+		); // changed
+
+		updateCustomTypeContentRelationships({
+			models: [{ model: getTwoLevelCrModel() }],
+			previousPath: ["author", "address_cr"],
+			newPath: ["author", "address_cr_CHANGED"],
+			onUpdate,
+		});
+
+		expect(onUpdate).toHaveBeenCalledWith(
+			getTwoLevelCrModel({ crId: "address_cr_CHANGED" }),
+		); // changed
+	});
+});
+
+describe("updateSharedSliceContentRelationships", () => {
+	it("should update slice content relationship ids", async () => {
+		const getOneLevelSharedSliceModel = (...ids: string[]) => {
+			return {
+				id: "testSlice",
+				name: "Test Slice",
+				type: "SharedSlice",
+				description: "Test Slice",
+				variations: [
+					{
+						id: "testVariation",
+						name: "Test Variation",
+						description: "Test Variation",
+						version: "1.0.0",
+						docURL: "https://www.prismic.io",
+						imageUrl: "https://www.prismic.io",
+						primary: {
+							name: {
+								type: "Text",
+								config: { label: "Name", placeholder: "Type name" },
+							},
+							authorDetails: {
+								type: "Link",
+								config: {
+									label: "Author Details",
+									placeholder: "Select author",
+									select: "document",
+									customtypes: [
+										{
+											id: "author",
+											fields: ["authorFirstName", ...ids],
+										},
+									],
+								},
+							},
+						},
+					},
+				],
+			} as const;
```

Comment:
:question: Is this `as const` needed?

Reply:
I would actually like to remove it, but for some reason I haven't yet figured out, `types-internal` imports are not being correctly resolved here, so I used `as const` to have the types matching.

Reply:
Currently testing typing the functions, to see if the CI also sees the import error of it it's just something in my local.

Reply:
Nope 🥲 I'm not seeing anything in the configuration that might cause this.

![Screenshot 2025-05-13 at 10 30 09](https://github.com/user-attachments/assets/be68d0af-157f-40e7-b44d-a4c20e0edf64)

Reply:
Just fixed it, looks like it needs a different import. Should be good!